### PR TITLE
- Maintenance task for polarization correction Fredrikze Algorithm

### LINKED ISSUE #38288: Maintenance tasks for PolarizationCorrectionFredrikze
Following on from https://github.com/mantidproject/mantid/pull/37672 we have identified the following maintenance that would be good to do for algorithm `PolarizationCorrectionFredrikze`:

- [x] Move the `PNR_LABEL` and `PA_LABEL` constants out of the `Prop` namespace and into the unnamed namespace below that holds all of the other label constants.
- [x] For the constants in the unnamed namespace, all should be of type `static const std::string`. The names should all be upper case with words separated by an underscore, so `crhoLabel` would become `CRHO_LABEL`, `cppLabel` would become `CPP_LABEL`, etc.
- [ ] Where we create the `spinStateValidator` by calling `std::make_shared<SpinStateValidator>(std::unordered_set<int>{2, 4}, true, 'p', 'a', true);` we should replace `'p'` and `'a'` with the constants `SpinStateConfigurationsFredrikze::PARA` and `SpinStateConfigurationsFredrikze::ANTI`.
- [x] In the algorithm documentation, the DOI in the References section at the bottom should be a hyperlink, but it doesn't seem to be rendering correctly.
- [x] In the algorithm documentation under the Spin State Configurations section, the note that gives the defaults of the `InputSpinStateOrder` property should probably be removed and that information should be integrated into the bullet points above. At the moment having a separate note makes the layout a bit confusing.

This final task should not be done until we have completed work on #38265:
- [x] We should make sure that existing unit tests are checking that the re-ordering of workspaces when different input and output spin states are passed in is being applied correctly. At the moment we seem to be assuming that as long as the algorithm completes without error then everything is OK. It may be that we can do this most effectively for the output spin state order after #38265. (Addressed as part of PR #38447)

### PR DESCRIPTION
**Pull request by yusufjimoh**
### Description of work
Move the PNR_LABEL and PA_LABEL constants out of the Prop namespace 
Rename some constants to use mantid naming convention 
updated documentation

One of the items in the referenced issue—replacing `'p'` and `'a'` with the constants `SpinStateConfigurationsFredrikze::PARA` and `SpinStateConfigurationsFredrikze::ANTI` in the creation of `spinStateValidator` using `std::make_shared<SpinStateValidator>(std::unordered_set<int>{2, 4}, true, 'p', 'a', true);`—could not be completed. This is because `SpinStateValidator` requires characters as inputs, but both constants are of type `std::string`. Additionally, these constants are used elsewhere in the code, making it difficult to change their type to characters.

Refs #38288 

### To test:
Follow the instructions on https://docs.mantidproject.org/nightly/algorithms/PolarizationCorrectionFredrikze-v1.html
unit tests should pass

*This does not require release notes because it is a maintenance task involving code changes

### CODE REVIEW

**Review by rbauststfc — COMMENTED** [review]
Thanks very much for fixing the DOI link at the bottom of the algorithm docs, that's working really well now. The changes are looking good, just a few things I noticed that could do with a bit of a look.

File: docs/source/algorithms/PolarizationCorrectionFredrikze-v1.rst [hunk 1]
```diff
@@ -32,13 +32,10 @@ The algorithm expresses the order of the workspaces in the input and output grou
 This property determines the order of the spin states in the input workspace group
 
 - :literal:`'pp, pa, ap, aa'`
-   Full polarization analysis (:literal:`'PA'`): both parallel, parallel-anti-parallel, anti-parallel-parallel, both  anti-parallel. Four input workspaces are required. The spin states can be provided in any order and should match the order of the workspaces in the input group.
+   Full polarization analysis (:literal:`'PA'`): both parallel, parallel-anti-parallel, anti-parallel-parallel, both  anti-parallel. Four input workspaces are required. The spin states can be provided in any order and should match the order of the workspaces in the input group. The default order is :literal:`pp, pa, ap, aa`
```
Comment (rbauststfc) [inline docs/source/algorithms/PolarizationCorrectionFredrikze-v1.rst:35]: Please could we add a full stop at the end of the final sentence in this paragraph?

File: docs/source/algorithms/PolarizationCorrectionFredrikze-v1.rst [hunk 2]
```diff
@@ -32,13 +32,10 @@ The algorithm expresses the order of the workspaces in the input and output grou
...
 
 - :literal:`'p, a'`
-   Polarized Neutron Reflectivity (:literal:`'PNR'`):  parallel, anti-parallel. Two input workspaces are required. The spin states can be provided in any order and should match the order of the workspaces in the input group.
-
-.. note::
-    the default order used by the algorithm if none is specified are :literal:`pp, pa, ap, aa` for full polarization analysis or :literal:`p, a` for Polarized Neutron Reflectivity.
+   Polarized Neutron Reflectivity (:literal:`'PNR'`):  parallel, anti-parallel. Two input workspaces are required. The spin states can be provided in any order and should match the order of the workspaces in the input group. The default order is :literal:`p, a`
```
Comment (rbauststfc) [inline docs/source/algorithms/PolarizationCorrectionFredrikze-v1.rst:38]: Please could we add a full stop at the end of the final sentence in this paragraph?

File: Framework/Algorithms/src/PolarizationCorrectionFredrikze.cpp [hunk 3]
```diff
@@ -25,8 +25,6 @@ using namespace Mantid::Kernel;
 using namespace Mantid::Geometry;
 
 namespace Prop {
-constexpr auto PNR_LABEL("PNR");
-constexpr auto PA_LABEL("PA");
 constexpr auto EFFICIENCIES{"Efficiencies"};
```
[the remark this answers is not in the document]
Comment (rbauststfc) [reply Framework/Algorithms/src/PolarizationCorrectionFredrikze.cpp:28]: It looks like this change wasn't included as part of your recent changes?

**Review by rbauststfc — COMMENTED** [review]
Thanks for making the constants in the unnamed namespace static. Looks like a couple of bits from my previous review haven't quite been picked up, so I've added some further comments if that's OK.

File: Framework/Algorithms/src/PolarizationCorrectionFredrikze.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (yusufjimoh) [reply Framework/Algorithms/src/PolarizationCorrectionFredrikze.cpp]: Noted, Thanks

**Review by rbauststfc — APPROVED** [review]
Thanks for this, everything is looking really good with the algorithm documentation and the constants are all of the correct type and named according to our naming conventions. All review comments have been addressed and this looks ready to go in.

**Pull request merged**